rpc/RPCSession.java

```diff
@@ -309,17 +374,17 @@ public boolean equals(Object o) {
                 if (o == null || getClass() != o.getClass()) return false;
                 Replica replica = (Replica) o;
                 return term == replica.term &&
-                        isLeader == replica.isLeader;
+                        isPrimary == replica.isPrimary;
             }
 
             @Override
             public int hashCode() {
-                return Objects.hash(isLeader, term);
+                return Objects.hash(isPrimary, term);
             }
 
             @Override
             public String toString() {
-                return id + ":" + (isLeader ? "L" : "N") + ":" + term;
+                return id + ":" + (isPrimary ? "P" : "S") + ":" + term;
```

Comment:
Not 100% sure if this is the right thing to do in the context; do you think we should expand these terms and use `Primary` and `Secondary` instead of `P` and `S`? I don't know how human-readable you want these to be.

Reply:
Yes, that's intentional. I intend to introduce a symbol that can describe a replica and its state in a concise way such that it can be easily included in the log without cluttering the whole line

```
[INFO] g.c.r.RPCSession - Replica '172.0.01:1729/grakn:P:5' is not responding after three attempts of reconnecting.
```

which would indicate that the replica of `grakn` which sits in the machine `172.0.01:1729`, is a primary replica of term 5 is not responding after three attempts.

They need to be short since we may need to print multiple of them in the same line given that we're working with multiple replicas most of the time.

I think exposing this information is highly useful not just for us but also for the users (eg., the app developers).

I think making them as short as possible (at the cost of being terse) is justified. What do you think @alexjpwalker ?

Reply:
Looks good to me 👍